add blog article about new client API (#975)
input-output-hk/mithril: docs/blog/2023-06-14-new-client-interface.md

---
title: Mithril client has got a brand new interface
authors:
- name: Mithril Team
tags: [client, certificate, mithril-stake-distribution]
---

> ## Mithril client now access signed Mithril Stake Distributions
> 
> Since the last few months, the Mithril team has been working on signing other artifacts than just Cardano nodes' snapshots. The first other signed artifact is the stake distribution involved in Mithril Signatures. The Aggregator nodes now advertise for different types of messages to be signed and the Signer nodes know if they can handle them or not without making old nodes to crash. If you are curious, you may go and see in [Mithril Explorer](https://mithril.network/explorer), there are now different tabs, one for each type of signed artifact.
> 
> The last part of this development was to make the signers aware of different kinds of artifacts and be able to verify theirs signature. This implied changing completely the command line API of the client command. By example:

Reviewer: I think the goal of this post is to warn users that the client interface will change shortly and that the way they retrieve snapshots has changed. This is not very clear in the sub title and the below introduction

```suggestion
### Mithril client interface is evolving

For the last few months, we have implemented the capability for the Mithril protocol to sign multiple types of data: on top of the already existing Cardano node database snapshots, the Mithril stake distribution is now also signed on its own. 

In order to make the client able to work on the different types of data that are certified, we have changed its command line API.

For example:
```

```

> $> mithril-client list

Reviewer: Also in the other commands :slightly_smiling_face: 
```suggestion
$> ./mithril-client list
```

```

> this command was previously used to list Cardano node snapshots. But now it is unclear that it lists snapshots, Mithril stake distributions or any other artifacts that will be supported in the future. So this has to be abandoned in favor of a more explicit syntax:

Reviewer: ```suggestion
This command was previously used to list Cardano node snapshots. It has been abandoned in favor of a more explicit syntax:
```

```
$> mithril-client snapshot list
```

Furthermore, the old version had two different subcommands to 1. download and 2. verify a snapshot. These 2 commands have now be merged into one single `download` command:

```

> $> mithril-client snapshot dowload  5109c1eaa6619bc…

Reviewer: ```suggestion
$> ./mithril-client snapshot dowload 5109c1eaa6619bc…
```

```

> This organization of the client opens the use of a new `mithril-stake-distribution` subcommand:

Reviewer: ```suggestion
This organization of the client opens the use of a new `mithril-stake-distribution` sub-command:
```

```
$> mithril-client mithril-stake-distribution list
```

Which can be aliased into a handy

```
$> mithril-client msd list
```

> In the same way it is for snapshots, it is possible to download and verify the stake distribution involved in Mithril signatures as JSON file:

Reviewer: ```suggestion
As for the Cardano snapshots, it is possible to download and verify the stake distribution involved in Mithril multi-signatures as a JSON file:
```

```
$> mithril-client msd download 713e2803e998f…
```

> If the file can be verified against the multi-signature, it is saved on the disk.

Reviewer: ```suggestion
If the file certification can be verified, it is saved on the disk.
```

> The Mithril stake distribution may not be a convenient artifact for Caradano users but it paves the way for other signed entities that will come in the future. You have an idea of something that you would like to see certified by Mithril multi-signature? Come and talk about it in our [Discord channel](https://discord.gg/5kaErDKDRq).

Reviewer: The conclusion is a bit confusing IMHO.
```suggestion

Feel free to reach out to us on the [Discord channel](https://discord.gg/5kaErDKDRq) for questions and/or help.

```
